libraries/plugins/witness/witness_plugin.cpp

```diff
@@ -465,14 +467,8 @@ namespace detail {
    }
 
    void witness_plugin_impl::schedule_production_loop() {
-      //Schedule for the next second's tick regardless of chain state
-      // If we would wait less than 50ms, wait for the whole second.
-      fc::time_point now = fc::time_point::now();
-      int64_t time_to_next_second = 1000000 - (now.time_since_epoch().count() % 1000000);
-      if(time_to_next_second < 50000)      // we must sleep for at least 50ms
-         time_to_next_second += 1000000;
-
-      _timer.expires_from_now( boost::posix_time::microseconds( time_to_next_second ) );
+      // Sleep for 200ms, before checking the block production
+      _timer.expires_from_now( boost::posix_time::microseconds( BLOCK_PRODUCTION_LOOP_SLEEP_TIME ) );
```

Comment:
Let's still keep the offset logic to of `now % 200000`. That way the thread doesn't drift over time. Because our blocks are sent to other nodes, it is important to try and wake on the second as often as possible to reduce block propagation latency.